commcare_connect/commcarehq_provider/provider.py

```diff
@@ -0,0 +1,33 @@
+from allauth.socialaccount import providers
+from allauth.socialaccount.providers.base import ProviderAccount
+from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
+
+
+class CommcareHQAccount(ProviderAccount):
+    def to_str(self):
+        return self.account.extra_data.get("name", super().to_str())
+
+
+class CommcareHQProvider(OAuth2Provider):
+    id = "commcarehq"
+    name = "commcarehq"
+    account_class = CommcareHQAccount
+
+    def get_default_scope(self):
+        return ["access_apis"]
+
+    def extract_uid(self, data):
+        return str(data["id"])
+
+    def extract_common_fields(self, data):
+        first_name = data.get("first_name")
+        last_name = data.get("last_name")
+        return dict(
+            email=data.get("email", ""),
+            name=" ".join([first_name, last_name]),
+            first_name=first_name,
+            last_name=last_name,
+        )
+
+
+providers.registry.register(CommcareHQProvider)
```

Comment:
This shouldn't be necessary. It should work if you have this:
```
provider_classes = [CommcareHQProvider]
```

Reply:
Fixed. [57067aa](https://github.com/dimagi/commcare-connect/pull/13/commits/57067aa21ede4d53afeb323eacaf20e11f3b9181)